crates/astria-composer/src/executor/tests.rs

```diff
@@ -481,3 +556,67 @@ async fn two_seq_actions_single_bundle() {
         );
     }
 }
+
+/// Test to check that executor's chain ID check is properly checked against the sequencer's chain
+/// ID
+#[tokio::test]
+async fn chain_id_mismatch_returns_error() {
+    use tendermint::chain::Id;
+
+    // set up sequencer mock
+    let (sequencer, cfg, _keyfile) = setup().await;
+    let shutdown_token = CancellationToken::new();
+    let metrics = Box::leak(Box::new(Metrics::new(cfg.parse_rollups().unwrap().keys())));
+
+    // mount a status response with an incorrect chain_id
+    mount_genesis(&sequencer, "bad-chain-id").await;
+
+    // build the executor with the correct chain_id
+    let (executor, _executor_handle) = executor::Builder {
+        sequencer_url: cfg.sequencer_url.clone(),
+        sequencer_chain_id: cfg.sequencer_chain_id.clone(),
+        private_key_file: cfg.private_key_file.clone(),
+        sequencer_address_prefix: cfg.sequencer_address_prefix.clone(),
+        block_time_ms: cfg.block_time_ms,
+        max_bytes_per_bundle: cfg.max_bytes_per_bundle,
+        bundle_queue_capacity: cfg.bundle_queue_capacity,
+        shutdown_token: shutdown_token.clone(),
+        metrics,
+    }
+    .build()
+    .unwrap();
+
+    // ensure that the chain id check function fails
+    let chain_id_check_err = executor
+        .ensure_chain_id_is_correct()
+        .await
+        .expect_err("executor::ensure_chain_id_is_correct() should return an error");
+    assert_chain_id_err(
+        &chain_id_check_err,
+        &cfg.sequencer_chain_id,
+        Id::try_from("bad-chain-id".to_string()).unwrap(),
+    );
```

Comment:
You can remove this block. You are already testing below, so this doesn't add anything of value.

Reply:
Done in 33ae83d.